src/tensor_ops/matmul/cpu_kernel.rs

```diff
@@ -71,14 +71,14 @@ impl MatMulImpl<half::f16> for Cpu {
                 m.size(),
                 n.size(),
                 k.size(),
-                cp as *mut gemm::f16,
+                cp,
                 cstr[1] as isize,
                 cstr[0] as isize,
                 accum,
-                ap as *const gemm::f16,
+                ap,
                 astr[1] as isize,
                 astr[0] as isize,
-                bp as *const gemm::f16,
+                bp,
```

Comment:
We still need these - internally gemm does a TypeId check against the type. So if gemm is using a different version of the half crate than dfdx, this will fail.